### ISSUE TITLE
Pattern: Table of contents - Accessibility - Currently active table of contents (Anchor ) links information is not conveyed for screen reader.

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/ibm-dotcom-library#1396]
## Detailed description
Device / browser - JAWS Firefox

Issue :- Once the pattern is loaded, the currently active link is visually highlighted. But, the same information is not conveyed for the screen reader user. 

Suggestion : The information conveyed through the visual presentation will not be identified by the screen reader. Hence the information should be made accessible using alternative methods. 
Here in this scenario, aria current attribute can be used for this purpose. Following link can be used for further reference. 
https://tink.uk/using-the-aria-current-attribute/

**Comment by RobertaJHahn** [comment]
@kennylam can you look at this issue and let us know whether this is a Sev 2 or not an issue?

**Comment by kennylam** [comment]
@RobertaJHahn @chetankan @Praveen-sr 
https://github.com/carbon-design-system/ibm-dotcom-library/pull/2354
